modules/apps/transfer/keeper/msg_server.go

```diff
@@ -59,3 +66,34 @@ func (k Keeper) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParams)
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// unwindHops unwinds the hops present in the tokens denomination and returns the message modified to reflect
+// the unwound path to take. It assumes that only a single token is present (as this is verified in ValidateBasic)
+// in the tokens list and ensures that the token is not native to the chain.
+func (k Keeper) unwindHops(ctx sdk.Context, msg *types.MsgTransfer) (*types.MsgTransfer, error) {
```

Comment:
nit: there's no need to return the msg when we're mutating a pointer, right?

func sig could just return error

Reply:
I think I'd prefer to not modify `msg` and return a new one entirely. Not a huge deal though